### PR TITLE
Add sleep binary to proxy image

### LINKED ISSUE #6723: WaitBeforeExitSeconds doesn't work due to missing /bin/bash inside proxy image
### What is the issue?
After swithicng to distroless base image for proxy (it was done in this [PR](https://github.com/linkerd/linkerd2/pull/6228)) `proxy.waitBeforeExitSeconds` functionality stoped working as image doesn't contain `/bin/bash` anymore.

### How can it be reproduced?
Configure `waitBeforeExitSeconds` to non zero value, and delete/update deployment and now it should be visible inside pod description or K8S node logs that pre stop hook failed due to missing `/bin/bash`.
```
I0823 10:28:58.312316    2346 handlers.go:67] Exec lifecycle hook ([/bin/bash -c sleep 25]) for Container "linkerd-proxy" in Pod "service-cos-5fd5469db9-l5zqr_default(17d0ac4f-4b1b-41cf-a8cd-9199226f2801)" failed - error: command '/bin/bash -c sleep 25' exited with 126: , message: "OCI runtime exec failed: exec failed: container_linux.go:380: starting container process caused: exec: \"/bin/bash\": stat /bin/bash: no such file or directory: unknown\r\n"
```
### Environment
- Linkerd version: linkerd-edge-21.8.1

### PR DESCRIPTION
**Pull request by adleong**
Fixes #6723

We add the sleep binary to the proxy image so that the waitBeforeExitSeconds will work.

### CODE REVIEW

File: Dockerfile-proxy [hunk 1]
```diff
@@ -37,6 +37,7 @@ COPY --from=fetch /build/proxy-version /usr/lib/linkerd/linkerd2-proxy-version.t
 COPY --from=fetch /build/linkerd2-proxy /usr/lib/linkerd/linkerd2-proxy
 COPY --from=fetch /build/linkerd-await /usr/lib/linkerd/linkerd-await
 COPY --from=golang /out/proxy-identity /usr/lib/linkerd/linkerd2-proxy-identity
+COPY --from=debian:buster-20210208-slim /bin/sleep /bin/sleep
```
Comment (alpeb) [inline Dockerfile-proxy:40]: we can go as recent as `bullseye-20210816-slim` here

**Review by alpeb — COMMENTED** [review]
This tested well for me :+1: Just a nit below

File: Dockerfile-proxy (hunk 1 above)
Comment (adleong) [reply Dockerfile-proxy:40]: buster-20210208-slim is used by the linkerd-cni container and the debug container.  I'm inclined to keep these on the same version for consistency and upgrade them all at once unless there's a reason not to.

**Review by kleimkuhler — APPROVED** [review]
😴

**Review by Pothulapati — APPROVED** [review]
LGTM :shipit:

**Comment by jasoares** [comment]
Hi, thanks for fixing this. Do you think we could have a Stable-2.10.3 release including this fix?

**Pull request merged**